python/adjoint/optimization_problem.py

```diff
@@ -69,12 +69,10 @@ def __init__(
         fcen=None,
         df=None,
         nf=None,
-        decay_dt=50,
-        decay_fields=[mp.Ez],
-        decay_by=1e-6,
+        decay_by=1e-11,
```

Comment:
Why does the default tolerance need to be reduced by five orders of magnitude? Won't such a small default value potentially lead to longer runtimes than are actually necessary particularly since the discretization errors are likely several orders of magnitude larger?

Reply:
The default tolerance now has a completely different meaning than before, so I'm not sure whether this actually corresponds to an increase in runtime…

Reply:
Exactly. This PR actually *reduces* the runtime from `master`, since it's able to check so frequently. But the tolerance corresponds to a completely different convergence check, which is why it's value is so different than before. 

Essentially the default tolerance is chosen to make the tests pass. But the runtime is still minimized (and less than `master`).